### PR TITLE
feat: add generic request for backend calls

### PR DESCRIPTION
**Pull request by aweiss-dev**
A generic way to access the backend and receive data via api-client

### CODE REVIEW

File: packages/api-client/src/generic/GenericAPI.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+import {HttpClient} from '../http';
+
+export class GenericAPI {
+  constructor(private readonly client: HttpClient) {}
+
+  /**
+   * Get Data from a given URL.
+   */
+  public async get<T = unknown>(url: string): Promise<T> {
```
Comment (atomrc) [inline packages/api-client/src/generic/GenericAPI.ts:30]: Don't we want something even more generic? Something like a `request` method that could do a GET/POST... and take a full `AxiosRequestConfig` object?

File: packages/api-client/src/generic/GenericAPI.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      url,
+    };
+
+    try {
+      const response = await this.client.sendJSON<T>(config);
+      return response.data;
+    } catch (error) {
+      throw error;
+    }
+  }
```
[the remark this answers is not in the document]
Comment (aweiss-dev) [reply packages/api-client/src/generic/GenericAPI.ts:42]: I was thinking about logging the error but removed it in the end. So yes, it is not needed anymore

File: packages/api-client/src/generic/GenericAPI.ts (hunk 1 above)
Comment (aweiss-dev) [reply packages/api-client/src/generic/GenericAPI.ts:30]: I would instead add post, put, and delete calls when needed. I was thinking about taking in a config, but I didn't have the use case for it yet.

Comment (atomrc) [reply packages/api-client/src/generic/GenericAPI.ts:30]: ok, fair enough, let's let it evolve as we go 👌

**Review by atomrc — APPROVED** [review]

**Pull request merged**